autotest/tools/restful/test_restful_chat_workspace.py

```diff
@@ -0,0 +1,197 @@
+from utils.run_restful_chat import (get_model, health_check, interactive_test,
+                                    open_chat_test)
+
+BASE_HTTP_URL = 'http://localhost'
+DEFAULT_PORT = 23333
+
+
+@pytest.fixture(scope='function', autouse=True)
+def prepare_environment(request, config, worker_id):
+    dst_path = config.get('dst_path')
+    log_path = config.get('log_path')
+
+    param = request.param
+    model = param['model']
+    cuda_prefix = param['cuda_prefix']
+    tp_num = param['tp_num']
+
+    if cuda_prefix is None:
+        cuda_prefix = get_cuda_prefix_by_workerid(worker_id, tp_num=tp_num)
+
+    worker_num = get_workerid(worker_id)
+    if worker_num is None:
+        port = DEFAULT_PORT
+    else:
+        port = DEFAULT_PORT + worker_num
+
+    cmd = get_command_with_extra('lmdeploy serve api_server ' + dst_path +
+                                 '/workspace_' + model + ' --server-port ' +
+                                 str(port),
+                                 config,
+                                 model,
+                                 need_tp=True,
+                                 cuda_prefix=cuda_prefix)
+
+    if 'kvint8' in model:
+        cmd += ' --quant-policy 4'
+        if 'w4' in model or '4bits' in model:
+            cmd += ' --model-format awq'
+        else:
+            cmd += ' --model-format hf'
+    if 'w4' in model or '4bits' in model:
+        cmd += ' --model-format awq'
+
+    start_log = os.path.join(log_path,
+                             'start_restful_' + model.split('/')[1] + '.log')
+
+    print('reproduce command restful: ' + cmd)
+
+    with open(start_log, 'w') as f:
+        f.writelines('reproduce command restful: ' + cmd + '\n')
+
+        # convert
+        convertRes = subprocess.Popen([cmd],
+                                      stdout=f,
+                                      stderr=f,
+                                      shell=True,
+                                      text=True,
+                                      encoding='utf-8')
+        pid = convertRes.pid
+    allure.attach.file(start_log, attachment_type=allure.attachment_type.TEXT)
+
+    http_url = BASE_HTTP_URL + ':' + str(port)
+    start_time = int(time())
+    sleep(5)
+    for i in range(120):
+        sleep(1)
+        end_time = int(time())
+        total_time = end_time - start_time
+        result = health_check(http_url)
+        if result or total_time >= 120:
+            break
+    yield
+    if pid > 0:
+        kill_log = os.path.join(log_path,
+                                'kill_' + model.split('/')[1] + '.log')
+
+        with open(kill_log, 'w') as f:
+            convertRes.kill()
+
+    allure.attach.file(kill_log, attachment_type=allure.attachment_type.TEXT)
+
+
+def getModelList(tp_num):
+    return [{
+        'model': item,
+        'cuda_prefix': None,
+        'tp_num': tp_num
+    } for item in get_turbomind_model_list(tp_num) if 'chat' in item.lower()]
```

Comment:
Will base models like `internlm2/internlm2-base-7b`  be tested?

Reply:
It's not covered yet. I will cover it with complation interface. not chat complation interface.